app/models/sheet.rb

```diff
@@ -3,16 +3,45 @@ class Sheet < ActiveRecord::Base
 
   validates :home_team, :away_team, :presence => true
 
-  after_initialize :init_boxes
+  before_create :init_boxes
+  before_create :generate_code
+  before_save :close_sheet, :if => Proc.new { |sheet| sheet.all_boxes_full? && !sheet.closed?}
+  before_save :set_team_score_arrays, :if => Proc.new { |sheet| sheet.closed? && sheet.team_score_arrays_blank? }
+
+  def set_team_score_arrays
+    self.home_team_score_row = random_score_array
+    self.away_team_score_row = random_score_array
+    self.save
+  end
+
+  def random_score_array
+    (0..9).to_a.shuffle
+  end
+
+  def all_boxes_full?
+    boxes.all? { |box| box.owner_name.present? }
```

Comment:
After thinking about this, we may need to add a `paid` attribute to each box.  Then calculate this was all `paid == true` for all boxes.  There might be a case where someone fills in the last box generating the random digits but not everyone has paid, inviting some tomfoolery lol